pkg/cloud/google/machineactuator.go

```diff
@@ -255,93 +255,109 @@ func (gce *GCEClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.Mach
 		return gce.handleMachineError(machine, verr, createEventAction)
 	}
 
```

Comment:
If we add a project field to machine provider spec we can remove the reliance on the cluster object as well if the machine was created via an instanceTemplate.

Reply:
I recall `project` was part of machine config once, but it was removed for some reasons (I don't know them, code/config duplication maybe?). In all other places (e.g. in machine actuator), reconciliation of Cluster API relies on `Cluster` anyway.